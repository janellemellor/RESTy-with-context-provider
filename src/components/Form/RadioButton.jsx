import React from 'react';
import PropTypes from 'prop-types';

export const RadioButtonGroup = ({ children, name, onInputChange }) => {
  const radioButtons = React.Children.map(children, child => {
    return React.cloneElement(child, {
      name,
      onInputChange 
    });
  });

  return (
    <>
      {radioButtons}
    </>
  );
};

export const RadioButton = ({ name, value, onInputChange }) => (
  <>
    <input type="radio" name={name} value={value} onInputChange={onInputChange} />
    <label htmlFor={value}>{value} </label>
  </>
);


RadioButtonGroup.propTypes = {
  children: PropTypes.node,
  name: PropTypes.string.isRequired,
  onInputChange: PropTypes.func.isRequired,
};

RadioButton.propTypes = {
  name: PropTypes.string.isRequired,
  onInputChange: PropTypes.func.isRequired,
  value: PropTypes.string
};

