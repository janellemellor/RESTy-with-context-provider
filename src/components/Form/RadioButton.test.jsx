import React from 'react';
import { shallow } from 'enzyme';
import { RadioButton } from './RadioButton';

describe('RadioButton Component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<RadioButton name="methods" value="POST" onInputChange={() =>{}} />);
    expect(wrapper).toMatchSnapshot();
  });    
});

