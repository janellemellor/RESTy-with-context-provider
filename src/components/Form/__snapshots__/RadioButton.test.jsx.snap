// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Component matches a snapshot 1`] = `
<Fragment>
  <input
    name="methods"
    onInputChange={[Function]}
    type="radio"
    value="POST"
  />
  <label
    htmlFor="POST"
  >
    POST
    &gt;
  </label>
</Fragment>
`;
